File: web/src/api/review.ts
/**
 * 层级六·AI 复盘 API 封装
 *
 * 对应后端 server/api/v1/review.py。复用 backtest.ts 的 apiClient。
 * 超时 90s：LLM 推理耗时较长，前端配合 loading 态。
 */
import { apiClient } from './backtest'

export interface ReviewRequest {
  csv_text?: string                    // 直接上传的日志文本（优先）
  start?: string                       // 或按日期读 logs/live_trades.csv
  end?: string
  strategy_name?: string
  strategy_params?: Record<string, unknown>
  metrics?: Record<string, unknown>    // 关键指标（max_drawdown 等）
}

export interface ReviewReport {
  ok: boolean
  report: string                       // Markdown 报告（或降级摘要）
  model?: string | null                // 实际模型；降级时 null
  degraded: boolean                    // LLM 不可用 → true
  reason?: string | null
}

/** 生成 AI 复盘报告（GLM；缺凭证时后端降级返回上下文摘要） */
export function diagnose(req: ReviewRequest): Promise<ReviewReport> {
  return apiClient.post('/api/v1/review/diagnose', req, { timeout: 90000 })
}

File: presentation/server/schemas/review.py
# -*- coding: utf-8 -*-
"""层级六·AI 复盘的 Pydantic 契约。

- ReviewRequest：复盘输入（csv_text 上传 或 start/end 按日期读日志 + 策略上下文）。
- ReviewReport：复盘输出（Markdown 报告；LLM 不可用时降级为上下文摘要）。
"""
from typing import Any, Dict, Optional

from pydantic import BaseModel


class ReviewRequest(BaseModel):
    """POST /review/diagnose 请求。

    数据源二选一：
    - csv_text：直接粘贴的实盘日志文本（CSV 格式，优先）。
    - start/end：按日期从 logs/live_trades.csv 读取（csv_text 缺省时生效）。
    策略上下文（可选，富化 Prompt）：
    - strategy_name / strategy_params / metrics（关键指标如 max_drawdown）。
    """
    csv_text: Optional[str] = None
    start: Optional[str] = None
    end: Optional[str] = None
    strategy_name: Optional[str] = None
    strategy_params: Dict[str, Any] = {}
    metrics: Dict[str, Any] = {}


class ReviewReport(BaseModel):
    """复盘报告响应。"""
    ok: bool
    report: str                           # Markdown 报告（或降级模式下的上下文摘要）
    model: Optional[str] = None           # 实际使用的模型（如 glm-4）；降级时为 None
    degraded: bool = False                # LLM 不可用（缺凭证/调用失败）→ True
    reason: Optional[str] = None          # 降级原因（degraded=True 时填）
